src/modules/assignment/assignment.service.ts

```diff
@@ -39,28 +65,68 @@ export class AssignmentService {
   async addAssignment(
     CreateAssignmentDTO: CreateAssignmentDTO,
   ): Promise<Assignment> {
-    const newAssignment = await new this.AssignmentModel(CreateAssignmentDTO);
-    return newAssignment.save();
+    try {
+      const newAssignment = await new this.AssignmentModel(CreateAssignmentDTO);
+      await newAssignment.save();
+      if (newAssignment) {
+        return newAssignment;
+      }
+      throw new NotFoundException('doubt not found!');
```

Comment:
No need for this, just return the saved assignments

Reply:
this is a different branch
PR#90 has the changes made